controller/cmd/service-mirror/metrics.go

```diff
@@ -0,0 +1,73 @@
+package servicemirror
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var labelNames = []string{
+	"gateway_name", "gateway_namespace", "remote_cluster_name",
+}
+
+type probeMetricVecs struct {
+	services  *prometheus.GaugeVec
+	alive     *prometheus.GaugeVec
+	latencies *prometheus.HistogramVec
+}
+
+type probeMetrics struct {
+	services  prometheus.Gauge
```

Comment:
Since we know the service name when it registers interest in the gateway, we could consider having a `service` label on this gauge.  This allows us to actually list all the services behind the gateway, instead of just knowing the number of services.

Reply:
Hm, fairly new to the whole Prometheus thing, but is it possible/practical to have multi value labels.  This metric is 1 per gateway, not per service. So how can we do what you are describing?

Reply:
I'm suggesting adding a label for service name. This would mean the gauge would only ever take values 1 (if the service with that name is served by that gateway) or 0 (if not).